components/director/internal/formationmapping/handler.go

```diff
@@ -137,6 +141,10 @@ func (h *Handler) updateFormationAssignmentStatus(w http.ResponseWriter, r *http
 	formation, err := h.formationService.Get(ctx, formationID)
 	if err != nil {
 		log.C(ctx).WithError(err).Errorf("An error occurred while getting formation from formation assignment with ID: %q", fa.FormationID)
+		if apperrors.IsNotFoundError(err) {
+			respondWithError(ctx, w, http.StatusBadRequest, errResp)
+			return
+		}
```

Comment:
the `errResp` is a "generic" err for the internal server errors. But with the new changes if the error is not found IMO I think the 404 is more appropriate(check the other comment as well) and the error message to the client should be adapted as well, e.g. "The formation assignment with ID: %s does not exist" or something similar.

Also, if we improve the status codes in the "status API" we should revisit all of the methods/responses and adapt them accordingly, e.g. 401, 404, 405, 500, 503, etc.

Reply:
I agree, we should revisit all responses. The ones I added here are so that we do not get alerts for 5xx requests when the error is on the client side. 
Currently if a user reports a status on the formations status API or formation assignment status API, it returns 500 in cases of not found. This often triggers the high number of 5xx requests alert and a person has to manually check if it is user error or an actual server error. It could also hide the actual 5xx requests.
This PR is more a slight improvement on the current situation, rather than an overhaul of what we return response and status code we return and when.

Reply:
I could change the new ones to 404 with a better message, as they should be.